setup.cfg

```diff
@@ -0,0 +1,12 @@
+[flake8]
+max-line-length = 110
+ignore = E133, E226, E228, E251, N802, N803, N806
+exclude = __init__.py, config/*.py, suprimecam/* hsc/*, bin/*
+
+[tool:pytest]
+addopts = --flake8
+flake8-ignore = E133 E226 E228 E251 N802 N803 N806
+    config/*.py ALL
+    suprimecam/* ALL
+    hsc/* ALL
+    bin/* ALL
```

Comment:
Is the `ALL` necessary? I have not seen it used before.
Also, why ignore everything in `bin`? I would expect all the python code to be from `bin.src` and to pass.

Reply:
Testing `bin/` and `bin.src/` doesn't add anything so I thought it was quicker to ignore them. The `ALL` might not be necessary, I didn't think it was until last week when a package I was working on with @SimonKrughoff required it. I'll take another look.

Reply:
I have experimented. See also https://pypi.org/project/pytest-flake8/

Specifying a file in the `flake8-ignore` section without any parameter does nothing. You need the `ALL` to prevent pytest from looking at the file at all. flake8 does respect the exclude line in the section above though. So without the ALL pytest sends the file to flake8 and flake8 returns a PASS because it was told to excliude the file. If you specify `ALL` then pytest never sends the file to flake8 so it doesn't even look like a test. `ALL` is therefore the correct approach.